pkg/workloadslicing/workloadslicing.go

```diff
@@ -66,26 +66,34 @@ func Enabled(object metav1.Object) bool {
 }
 
 const (
-	// WorkloadSliceReplacementFor is the annotation key used to capture an "old" workload slice key
-	// that will be preempted by the "new", e.g., this workload slice with annotation.
+	// WorkloadSliceReplacementFor is the annotation key set on a new workload slice to indicate
+	// the key of the workload slice it is intended to replace (i.e., the "old" slice being preempted).
 	WorkloadSliceReplacementFor = "kueue.x-k8s.io/workload-slice-replacement-for"
+
+	// WorkloadSliceClusterName is the annotation key set on a new workload slice to record
+	// the name of the remote cluster where the original workload was scheduled in a multikueue context.
+	// This annotation can be used to enforce workload propagation or affinity policies.
+	WorkloadSliceClusterName = "kueue.x-k8s.io/workload-slice-cluster-name"
 )
 
 // ReplacementForKey returns a value for workload "WorkloadSliceReplacementFor" annotation
-// key if this workload was annotated with such, otherwise, returns an empty string.
 func ReplacementForKey(wl *kueue.Workload) *workload.Reference {
-	annotations := wl.GetAnnotations()
-	if len(annotations) == 0 {
-		return nil
-	}
-	key, found := annotations[WorkloadSliceReplacementFor]
+	key, found := wl.GetAnnotations()[WorkloadSliceReplacementFor]
 	if !found {
 		return nil
 	}
 	ref := workload.Reference(key)
 	return &ref
 }
 
+// ClusterName returns the name of the remote cluster where the original workload
+// was scheduled in a multikueue context. If the corresponding annotation is not set,
+// it returns an empty string.
+func ClusterName(wl *kueue.Workload) string {
+	return wl.GetAnnotations()[WorkloadSliceClusterName]
```

Comment:
If we need to add a different cluster nomination mechanism from non-WorkloadSlice usage, I would request updating KEP-77.